subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -166,15 +167,42 @@ public String determineName(Configuration configuration) {
     Configuration setExtendsFrom(Iterable<Configuration> superConfigs);
 
     /**
-     * Adds the given configurations to the set of configuration which this configuration extends from.
+     * Adds the configuration from the given provider to the set of configuration which this configuration extends from.
      * <p>
      * Configurations are only allowed to extend from other configurations in the same project.
      *
+     * @param configProvider Provider of the super configuration.
+     * @return this configuration
+     *
+     * @since 8.11
+     */
+    @Incubating
+    default Configuration setExtendsFrom(Provider<Configuration> configProvider) {
+        return setExtendsFrom(Collections.singletonList(configProvider.get()));
```

Comment:
These changes immediately calculates the provider. Is this expected? If so, it should be mentioned in the docs.

Reply:
Yes, we don't intend to support lazy extension, this is just a bit of porcelain for the API.  

Good idea to update the documentation, I'll do that.